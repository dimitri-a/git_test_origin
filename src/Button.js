const styled = styled.default;

const Button = styled.button`
	background: ${props => props.primary ? 'palevioletred' : 'white'};
	color: ${props => props.primary ? 'white' : 'palevioletred'};

	font-size: 1em;
	margin: 1em;
	padding: 0.25em 1em;
	border: 2px solid palevioletred;
	border-radius: 3px;

  &:hover {
  ${Button} > {
    background-color: green;
  }
}
`;

