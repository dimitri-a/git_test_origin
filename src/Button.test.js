import React from 'react'

import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Button } from './Button'
import {Button2} from "./Button2";

test('button test', () => {
    const tree = renderer.create(<Button primary /> ).toJSON()
    expect(tree).toMatchSnapshot()
    console.log(tree);
    expect(tree).toHaveStyleRule('color', 'white')
    expect(tree).toHaveStyleRule('background', 'red')
})