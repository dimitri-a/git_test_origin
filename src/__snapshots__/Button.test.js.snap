// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button test 1`] = `
.c0 {
  background: red;
  color: white;
  font-size: 1em;
  margin: 1em;
  padding: 0.25em 1em;
  border: 2px solid palevioletred;
  border-radius: 3px;
}

<button
  className="c0"
/>
`;
